transmogrifier/sources/xml/dspace_mets.py

```diff
@@ -212,6 +136,136 @@ def get_content_type(cls, source_record: Tag) -> list[str] | None:
             for content_type in source_record.find_all("mods:genre", string=True)
         ] or None
 
+    @classmethod
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor] | None:
+        contributors = []
+        for contributor in source_record.find_all("mods:name"):
+            if name := contributor.find("mods:namePart", string=True):
+                if role := contributor.find("mods:roleTerm", string=True):
+                    kind = str(role.string)
+                else:
+                    kind = "Not specified"
+                contributors.append(
+                    timdex.Contributor(
+                        kind=kind,
+                        value=str(name.string),
+                    )
+                )
+        return contributors or None
+
+    @classmethod
+    def get_dates(cls, source_record: Tag) -> list[timdex.Date] | None:
+        # Only publication date is mapped from DSpace, other relevant date field (dc.
+        # coverage.temporal) is not mapped to the OAI-PMH METS output.
```

Comment:
There are several of these comments that were copied over, not sure how useful they really are and they feel like clutter to me but wanted to preserve them for your review. There's also this [ticket](https://mitlibraries.atlassian.net/browse/TIMX-16) which could render this entire transform unnecessary should we decide to go with `DspaceDim` for the `DSpace@MIT` instance

Reply:
Ignoring for the moment if we keep `DSpaceMETS`, I think they'd make nice docstrings vs standalone comments.  To me, this is a big benefit of the field methods: a place for this kind of documentation.  [Here's an example from GeoHarvester](https://github.com/MITLibraries/geo-harvester/blob/main/harvester/records/formats/fgdc.py#L291-L297).  The rub is that our linting requires a first line title (short, ending in period) and then can have freeform text.  My two cents is that a fairly generic title like "Field method: <field name>" is okay, then can launch into commentary.

I think it's nice to keep them for now, but thanks for sharing the ticket, that it may be moot eventually.

Reply:
Good call, I shifted them over!